packages/datadog-plugin-aws-sdk/test/kinesis.spec.js

```diff
@@ -35,114 +48,192 @@ describe('Kinesis', () => {
-        expect(span.meta).to.include({
-          'streamname': 'MyStream',
-          'aws_service': 'Kinesis',
-          'region': 'us-east-1'
+      it('generates tags for proper input', done => {
+        agent.use(traces => {
+          const span = traces[0][0]
+          expect(span.meta).to.include({
+            'streamname': streamName,
+            'aws_service': 'Kinesis',
+            'region': 'us-east-1'
+          })
+          expect(span.resource).to.equal(`putRecord ${streamName}`)
+          expect(span.meta).to.have.property('streamname', streamName)
+        }).then(done, done)
+
+        helpers.putTestRecord(kinesis, streamName, helpers.dataBuffer, e => e && done(e))
+      })
+
+      describe('Disabled', () => {
+        before(() => {
+          process.env.DD_TRACE_AWS_SDK_KINESIS_ENABLED = 'false'
+        })
+
+        after(() => {
+          delete process.env.DD_TRACE_AWS_SDK_KINESIS_ENABLED
         })
-        expect(span.resource).to.equal('putRecord MyStream')
-        expect(span.meta).to.have.property('streamname', 'MyStream')
-      }).then(done, done)
 
-      helpers.putTestRecord(kinesis, helpers.dataBuffer, e => e && done(e))
+        it('skip injects trace context to Kinesis putRecord when disabled', done => {
+          helpers.putTestRecord(kinesis, streamName, helpers.dataBuffer, (err, data) => {
+            if (err) return done(err)
+
+            helpers.getTestData(kinesis, streamName, data, (err, data) => {
+              if (err) return done(err)
+
+              expect(data).not.to.have.property('_datadog')
+
+              done()
+            })
+          })
+        })
+      })
     })
 
-    describe('Disabled', () => {
+    describe('DSM Context Propagation', () => {
       before(() => {
-        process.env.DD_TRACE_AWS_SDK_KINESIS_ENABLED = 'false'
+        return agent.load('aws-sdk', { kinesis: { dsmEnabled: true } }, { dsmEnabled: true })
       })
 
-      after(() => {
-        delete process.env.DD_TRACE_AWS_SDK_KINESIS_ENABLED
+      before(done => {
+        tracer = require('../../dd-trace')
+        tracer.use('aws-sdk', { kinesis: { dsmEnabled: true } }, { dsmEnabled: true })
+
+        createResources(streamNameDSM, done)
+      })
+
+      after(done => {
+        kinesis.deleteStream({
+          StreamName: streamNameDSM
+        }, (err, res) => {
+          if (err) return done(err)
+
+          helpers.waitForDeletedStream(kinesis, streamNameDSM, done)
+        })
       })
 
-      it('skip injects trace context to Kinesis putRecord when disabled', done => {
-        helpers.putTestRecord(kinesis, helpers.dataBuffer, (err, data) => {
+      it('injects DSM trace context to Kinesis putRecord', done => {
+        if (DataStreamsContext.setDataStreamsContext.isSinonProxy) {
+          DataStreamsContext.setDataStreamsContext.restore()
+        }
+        const setDataStreamsContextSpy = sinon.spy(DataStreamsContext, 'setDataStreamsContext')
```

Comment:
Plugin tests are integration tests and should only validate input/output and not implementation details. Can this (and other spies from this PR) be done as integration tests instead?

Reply:
Please re review, I updated the tests to no longer use spies.